src/vs/workbench/browser/quickaccess.ts

```diff
@@ -45,3 +47,44 @@ export function getQuickNavigateHandler(id: string, next?: boolean): ICommandHan
 		quickInputService.navigate(!!next, quickNavigate);
 	};
 }
+
+export function getFileSelectionTextFromEditor(allowUnselectedWord: boolean, editor: IEditor): string | null {
+
+	if (!isCodeEditor(editor) || !editor.hasModel()) {
+		return null;
+	}
+
+	const range = editor.getSelection();
+	if (!range) {
+		return null;
+	}
+
+	if (range.isEmpty()) {
+		if (allowUnselectedWord) {
+			const wordAtPosition = editor.getModel().getWordAtPosition(range.getStartPosition());
+			return wordAtPosition?.word ?? null;
+		} else {
+			return null;
+		}
+	}
+
+	let searchText = '';
+	for (let i = range.startLineNumber; i <= range.endLineNumber; i++) {
+		let lineText = editor.getModel().getLineContent(i);
+		if (i === range.endLineNumber) {
+			lineText = lineText.substring(0, range.endColumn - 1);
+		}
+
+		if (i === range.startLineNumber) {
+			lineText = lineText.substring(range.startColumn - 1);
+		}
+
+		if (i !== range.startLineNumber) {
+			lineText = '\n' + lineText;
+		}
+
+		searchText += lineText;
+	}
+
+	return searchText;
+}
```

Comment:
just noticed this, but I think that https://github.com/microsoft/vscode/blob/a7a4d6a72ddda3f1ee463da1d099a43517a5903c/src/vs/workbench/contrib/search/browser/searchView.ts#L2143-L2192

has pretty much the same code, but also makes sure to get text from diff editors. Can you reuse that?

Reply:
I tried that earlier, I get the following error if I do that while committing 
Imports violates 'vs/css!./**/* or vs/base/{common,browser}/** or vs/base/parts/*/{common,browser}/** or vs/platform/*/{common,browser}/** or vs/editor/{common,browser}/** or vs/editor/contrib/*/{common,browser}/** or vs/workbench/{common,browser}/** or vs/workbench/services/*/{common,browser}/** or assert or vs/nls or vs/amdX' restrictions.

Its not letting me import anything from the contrib folder.
 https://github.com/microsoft/vscode/wiki/Source-Code-Organization

Reply:
One way I can think of is writing this function somewhere, probably a folder from which is import is allowed. And then reusing it in both the places. Not sure exactly where though. Are there any conventions?

Reply:
I'm okay with this code living in `quickaccess.ts` as long as you have the most up-to-date version of the function (including the newest changes from today). 
@TylerLeonhardt are you okay with the selection-grabbing code being here and search just importing it? Ideally, the selection code could be somewhere else more communal in the future and be shared by everything that gets editor selection.

Reply:
Shall I import the function from quickaccess.ts to searchView.ts then? I think this import is possible. I can change the function name from getFileSelectionTextFromEditor to getSelectionTextFromEditor

Reply:
Yes, the function would only occur once in `quickAccess.ts`. Make sure that you have the part 

```
 	if (isDiffEditor(editor)) { 
 		if (editor.getOriginalEditor().hasTextFocus()) { 
 			editor = editor.getOriginalEditor(); 
 		} else { 
 			editor = editor.getModifiedEditor(); 
 		} 
 	} 
  ```
Since it looks like your current implementation doesn't have it and it's needed to get the selection in diff editors.

Reply:
Sure, I will add that too

Reply:
Yes I think having this in `quickaccess.ts` for this and search to use makes sense.

Reply:
I updated the PR with those changes, let me know if this is correct!